catalog/entry_catalog.go

```diff
@@ -173,6 +174,16 @@ func (e *EntryCatalog) CreateRepository(ctx context.Context, repositoryID gravel
 	return e.Store.CreateRepository(ctx, repositoryID, storageNamespace, branchID)
 }
 
+func (e *EntryCatalog) CreateBareRepository(ctx context.Context, repositoryID graveler.RepositoryID, storageNamespace graveler.StorageNamespace, branchID graveler.BranchID) (*graveler.Repository, error) {
+	if err := Validate([]ValidateArg{
+		{"repositoryID", repositoryID, ValidateRepositoryID},
+		{"storageNamespace", storageNamespace, ValidateStorageNamespace},
+	}); err != nil {
+		return nil, err
+	}
+	return e.Store.CreateBareRepository(ctx, repositoryID, storageNamespace, branchID)
```

Comment:
This is one of multiple copies of `CreateRepository` => `CreateBareRepository`.  It seems like a lot more work than passing another boolean parameter, and/or splitting `CreateRepository` into `CreateBareRepository` + `SomethingRepositoryDefaultBranchAndCommit` higher up the call tree.

Reply:
Yea, still not sure how to handle "plumbing" calls in our internal APIs. Open to concrete suggestions (i'm fine with all options you suggested above)